components/deferComponentRender.js

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) 2017 Mattermost Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import hoistStatics from 'hoist-non-react-statics';
+import React from 'react';
+
+/**
+ * Allows two animation frames to complete to allow other components to update
+ * and re-render before mounting and rendering an expensive `WrappedComponent`.
+ * If provided, `PreRenderComponent` will be rendered instead of null when not
+ * rendering the `Wrapped Component`.
+ *
+ * Based on this Twitter built component
+ * https://gist.github.com/paularmstrong/cc2ead7e2a0dec37d8b2096fc8d85759#file-defercomponentrender-js
+ */
+export default function deferComponentRender(WrappedComponent, PreRenderComponent = null) {
+    class DeferredRenderWrapper extends React.Component {
+        constructor(props, context) {
+            super(props, context);
+
+            this.state = {
+                shouldRender: false
+            };
+        }
+
+        componentDidMount() {
+            window.requestAnimationFrame(() => {
+                window.requestAnimationFrame(() => this.setState({ shouldRender: true }));
+            });
+        }
```

Comment:
I was trying to think of another way to write it without the nesting but I don't think I simplified the code:
```javascript
async componentDidMount() {
    await window.requestAnimationFrame(() => Promise.resolve());
    await window.requestAnimationFrame(() => Promise.resolve());
    this.setState({ shouldRender: true });
}
```
What you have is 👍